consensus/service/src/mint_tx_manager/mod.rs

```diff
@@ -176,48 +167,39 @@ impl<L: Ledger> MintTxManager for MintTxManagerImpl<L> {
 
     fn combine_mint_txs(
         &self,
-        txs: &[MintTx],
+        txs: &[(MintTx, u64)],
         max_elements: usize,
-    ) -> MintTxManagerResult<Vec<MintTx>> {
-        let mut candidates = txs.to_vec();
-        candidates.sort();
+    ) -> MintTxManagerResult<Vec<(MintTx, u64)>> {
+        let candidates = timestamp_validator::sort_and_dedup(txs.iter());
 
-        let mut seen_nonces = HashSet::default();
         let mut seen_mint_configs = HashSet::default();
-        let (allowed_txs, _rejected_txs) = candidates.into_iter().partition(|tx| {
-            let nonce_with_token_id = NonceByTokenId {
-                nonce: tx.prefix.nonce.clone(),
-                token_id: tx.prefix.token_id,
-            };
-            if seen_nonces.len() >= max_elements {
-                return false;
-            }
-            if seen_nonces.contains(&nonce_with_token_id) {
-                return false;
-            }
-
-            // We allow a specific MintConfig to be used only once per block - this
-            // simplifies enforcing the total mint limit.
-            let active_mint_config = match self.ledger_db.get_active_mint_config_for_mint_tx(tx) {
-                Ok(active_mint_config) => active_mint_config,
-                Err(err) => {
-                    log::warn!(
-                        self.logger,
-                        "failed finding an active mint config for mint tx {}: {}",
-                        tx,
-                        err
-                    );
+        let allowed_txs = candidates
```

Comment:
Similar to MintTxConfig, notice how this looks at the entire `MintTx` for deduplication over only the nonce and token id

Reply:
brought back logic to prevent multiple `(nonce, token_id)`s in the mint transactions.